File: src/main/java/com/sfh/pokeRogueBot/phase/impl/CheckSwitchPhase.java
package com.sfh.pokeRogueBot.phase.impl;

import com.sfh.pokeRogueBot.model.enums.UiMode;
import com.sfh.pokeRogueBot.model.exception.NotSupportedException;
import com.sfh.pokeRogueBot.phase.AbstractPhase;
import com.sfh.pokeRogueBot.phase.Phase;
import com.sfh.pokeRogueBot.phase.actions.PhaseAction;
import com.sfh.pokeRogueBot.service.Brain;
import org.springframework.stereotype.Component;

@Component
public class CheckSwitchPhase extends AbstractPhase implements Phase {

    public static final String NAME = "CheckSwitchPhase";

    private final Brain brain;

    public CheckSwitchPhase(Brain brain) {
        this.brain = brain;
    }

    @Override
    public String getPhaseName() {
        return NAME;
    }

    @Override
    public PhaseAction[] getActionsForGameMode(UiMode gameMode) throws NotSupportedException {
        if (gameMode == UiMode.CONFIRM) {

            boolean shouldSwitchPokemon = brain.shouldSwitchPokemon();

            if(shouldSwitchPokemon) {
                return new PhaseAction[]{
                        this.pressSpace
                };
            }
            else {
                return new PhaseAction[]{
                        this.pressArrowDown,
                        this.waitBriefly,
                        this.pressSpace
                };
            }

        } else if (gameMode == UiMode.MESSAGE) {
            return new PhaseAction[]{
                    this.waitLonger
            };
        } else {
            throw new NotSupportedException("GameMode not supported: " + gameMode);
        }
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/phase/impl/CheckSwitchPhase.kt
package com.sfh.pokeRogueBot.phase.impl

import com.sfh.pokeRogueBot.model.enums.UiMode
import com.sfh.pokeRogueBot.model.exception.UiModeException
import com.sfh.pokeRogueBot.phase.UiPhase
import com.sfh.pokeRogueBot.service.Brain
import com.sfh.pokeRogueBot.service.javascript.JsUiService
import org.springframework.stereotype.Component

@Component
class CheckSwitchPhase(
    private val brain: Brain,
    private val jsUiService: JsUiService
) : UiPhase {

    override val phaseName: String = "CheckSwitchPhase"

    override fun handleUiMode(uiMode: UiMode) {
        if (uiMode == UiMode.CONFIRM) {
            val shouldSwitchPokemon = brain.shouldSwitchPokemon()

            if (shouldSwitchPokemon) {
                jsUiService.sendActionButton()
                return
            } else {
                jsUiService.setUiHandlerCursor(uiMode, 1) //set to no
                jsUiService.sendActionButton()
                return
            }
        }

        throw UiModeException(uiMode)
    }
}